ibis/expr/tests/test_reductions.py

```diff
@@ -161,3 +162,22 @@ def test_ordered_aggregations_no_order(method):
     q3 = func(order_by=())
     assert q1.equals(q2)
     assert q1.equals(q3)
+
+
+def test_collect_distinct():
+    t = ibis.table({"a": "string", "b": "int", "c": "int"}, name="t")
+    # Fine
+    t.a.collect(distinct=True)
+    t.a.collect(distinct=True, order_by=t.a.desc())
+    (t.a + 1).collect(distinct=True, order_by=(t.a + 1).desc())
```

Comment:
Are these necessary if we're testing them in an execution test?

Reply:
This test is for checking the validation check on `order_by`, I want to have decent coverage here in a single test. We do hit one of these cases in the execution test, but that's more for testing execution is correct, not the expression validation.